packages/edit-widgets/src/components/customizer-layout/index.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	SlotFillProvider,
+	Popover,
+	navigateRegions,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import WidgetAreas from '../widget-areas';
+import { withWPCustomize } from '../../utils';
+
+function CustomizerLayout( { blockEditorSettings } ) {
+	return (
+		<SlotFillProvider>
+			<div
+				className="edit-widgets-customizer-layout__content"
+				role="region"
+				aria-label={ __( 'Widgets screen content' ) }
+				tabIndex="-1"
+			>
+				<WidgetAreas
+					blockEditorSettings={ blockEditorSettings }
+					onUpdateBlocks={ useCallback(
+						( id, blocks ) =>
+							withWPCustomize( ( { previewBlocksInWidgetArea, saveButton } ) => {
```

Comment:
I guess customizer was added because for customizer we need to call previewBlocksInWidgetArea. What if instead on our customizer code we subscribe to the widgets store `wp.data.subscribe('core/edit-widgets')` and then when an area changes we update its preview? I guess it would also provide more isolation as we would not have the widgets module calling/interacting with the dom/external code, instead external code would react to changes in the widget store.
Do you think it would simplify things or do you predict some disadvantage?

Reply:
It's also needed for styling right now. We could also just have the extra class toggled through a prop, but I wouldn't be surprised if we do more layout changes on the widgets screen that shouldn't also affect the Customizer section.

That subscription approach would work, although you would have to subscribe to the whole registry (it only supports global listeners), run the relevant selectors, and call a memoized function with the blocks. All in all, that makes the code way more complex. I don't think unavoidable DOM manipulations like these in a React tree are bad as long as they are minimized and only run on side effects or interactions. But, if we have a pre existing convention in the rest of the codebase, I can change it.

Reply:
I ended up going with the manual subscription approach, because the complexity that came with adding scheduling, drafting, and share URLs warranted it.